enterprise/frontend/src/embedding-sdk/cli/types/types.ts

```diff
@@ -0,0 +1,48 @@
+export type CliState = Partial<{
+  port: number;
+  instanceUrl: string;
+  cookie: string;
+  apiKey: string;
+  email: string;
+  password: string;
+  token: string;
+}>;
+
+export type CliError = {
+  type: "error";
+  message: string;
+};
+
+export type CliSuccess = {
+  type: "success";
+  nextStep?: CliStep;
+};
+
+export type CliDone = {
+  type: "done";
+};
+
+export type CliStepType = CliError | CliSuccess | CliDone;
+
+export type CliOutput = [CliStepType, CliState];
+
+export type CliStepMethod = (
+  state: CliState,
+  ...options: any[]
+) => CliOutput | Promise<CliOutput>;
+
+export const CLI_ORDER = [
```

Comment:
maybe move this to `constants`?

Reply:
@oisincoveney yes, best to not have constants in types 👍🏻